### DISCUSSION TITLE
Support for running %check on a different machine

### DISCUSSION THREAD

**Discussion by DemiMarie** [rpm-software-management/rpm#2884] [Ideas]
To the best of my understanding, cross-compilation currently is incompatible with running test suites, unless one uses QEMU or other pure-software emulation mechanisms.  These cause problems with code that depends on CPU features.

At the same time, the rise of languages such as Rust and C++, which require more CPU and RAM during build, is making it very difficult to natively compile on slow processors.

One potential solution would be for RPM to support running the test suite remotely, or at least in a separate step.

**Comment by pmatilai** [comment]
@dmnks feel free to correct/extend, but the test-suite does run as a separate step in the "native" mode using already built binaries. That's only supported on Fedora at the moment but that's AIUI just a matter of somebody adding + maintaining a native dockerfile for their target distro.

**Comment by DemiMarie** [comment]
@pmatilai I was referring to `%check` in spec files, _not_ RPM’s own test suite (though that would be useful too).

**Comment by pmatilai** [comment]
Oh. Reading more carefully now, I can see a lack of "the" and a relevant plural in "running test suites"  :sweat_smile: Adjusted the title to make it clear.

All the build scriptlets have (some) support for remote execution through this:

> %___build_cmd           %{?_sudo:%{_sudo} }%{?_remsh:%{_remsh} %{_remhost} }%{?_remsudo:%{_remsudo} }%{?_remchroot:%{_remchroot} %{_remroot} }%{___build_shell} %{___build_args}

It requires setting up some shared mounts of course (for %_tmpdir and the build area at least), I've never tried that in practise.
